File: src/arduinobot_cpp_examples/src/simple_subscriber.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>
#include <string>

using std::placeholders::_1;

class SimpleSubscriber : public rclcpp::Node
{
    public:
        SimpleSubscriber() : Node("simple_subscriber")
        {
            subscriber_ = create_subscription<std_msgs::msg::String>("example_topix", 10, 
                std::bind(&SimpleSubscriber::OnMessageReceived, this, _1));
        }

        void OnMessageReceived(const std_msgs::msg::String::SharedPtr msg) const
        {
            RCLCPP_INFO(this->get_logger(), "Received message: '%s'", msg->data.c_str());
        }

    private:
        rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscriber_;

};

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SimpleSubscriber>());
    rclcpp::shutdown();
    return 0;
}

File: src/arduinobot_py_examples/arduinobot_py_examples/simple_subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimpleSubscriber(Node):
    def __init__(self):
        super().__init__('simple_subscriber')
        self.subcriber_ = self.create_subscription(String,'example_topix', self.onMessageReceived, 10)
        self.get_logger().info('Subscribed to ''example_topix''')

    def onMessageReceived(self, msg):
        self.get_logger().info(f'Received message: {msg.data}')

def main(args=None):
    rclpy.init()
    simpleSubscriber = SimpleSubscriber()
    rclpy.spin(simpleSubscriber)
    simpleSubscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()